### PR TITLE
feat: Add team filter in timeline history page

### PR DESCRIPTION
**Pull request by tianrunhe**
# Description
Added team filter in timeline history page

## Testing scenarios

- [ ] Team filter in history page
  - Go to history page
  - Select a team from the team filter dropdown menu
  - Observe the events are filtered for that team
  - Verify infinity scrolling still work

- [ ] Combination of team filter & event type filter
  - Go to history page
  - Select a team from the team filter dropdown menu, and then select an event type from the event filter dropdown menu
  - Verify the events are filtered
  - Verify you can "reset" by selecting "All Teams" & "All Events"

- [ ] Empty State
  - Go to history page
  - Select filters that do not match to any events
  - Observe the empty state message shows up and you can reset back to "All Teams" & "All Events"

## Final checklist

- [x] I checked the [code review guidelines](../docs/codeReview.md)
- [x] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [x] I have performed a self-review of my code, the same way I'd do it for any other team member
- [x] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [x] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [x] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [x] PR title is human readable and could be used in changelog

### CODE REVIEW

File: packages/server/graphql/getValidTeamIds.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const getValidTeamIds = (teamIds: null | string[], tms: string[]) => {
+  // the following comments can be removed pending #4070
+  // const viewerTeamMembers = await dataLoader.get('teamMembersByUserId').load(viewerId)
+  // const viewerTeamIds = viewerTeamMembers.map(({teamId}) => teamId)
+  if (teamIds?.length) return teamIds!.filter((teamId) => tms.includes(teamId))
```
Comment (nickoferrall) [inline packages/server/graphql/getValidTeamIds.ts:5]: +1 this isn't your code, but we can remove the `!` from `teamIds!` as we're checking if `teamIds` exists

File: packages/server/graphql/getValidTeamIds.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const getValidTeamIds = (teamIds: null | string[], tms: string[]) => {
+  // the following comments can be removed pending #4070
```
Comment (nickoferrall) [inline packages/server/graphql/getValidTeamIds.ts:2]: +1 what's #4070 referring to? This: https://github.com/ParabolInc/parabol/issues/4070? 

If so, it looks like we can remove the commented lines

**Review by nickoferrall — APPROVED** [review]

File: packages/server/graphql/getValidTeamIds.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import {DataLoaderWorker} from './graphql'
+
+export const getValidTeamIds = async (
```
Comment (Dschoordsch) [inline packages/server/graphql/getValidTeamIds.ts:3]: -1 `getAccessibleTeamIds` or `filterByTeamMemberIds` would be more descriptive. All input ids could be "valid" in other contexts.

File: packages/server/graphql/getValidTeamIds.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  viewerId: string,
+  teamIds: null | string[],
+  dataLoader: DataLoaderWorker
+) => {
+  const viewerTeamMembers = await dataLoader.get('teamMembersByUserId').load(viewerId)
+  const viewerTeamIds = viewerTeamMembers.map(({teamId}) => teamId)
+  if (teamIds?.length) return teamIds.filter((teamId) => viewerTeamIds.includes(teamId))
+  // filter the teamIds array to only teams the user has a team member for
```
Comment (Dschoordsch) [inline packages/server/graphql/getValidTeamIds.ts:11]: +1 this could be the function comment

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
Only a naming issue, otherwise looks good!
Feel free to merge when fixed.

**Review by Dschoordsch — APPROVED** [review]

**Pull request merged**